### DISCUSSION TITLE
Vectorize structured data

### DISCUSSION THREAD

**Discussion by gtiquant** [milvus-io/milvus#27260] [Q&A and General discussion]
I have a question. Is it good idea to vectorize structure relational data that is stored in firebase firestore db. So that I can QA over that data.
I need some advice.

For example my sample data schema is:
////////////
Entity: users
Fields: {user_id, name, phone, email}

Entity: food_items
Fields: {item_id, name, price}

Entity: orders
Fields: {order_id, {items array containing food item id, price and quantity ordered}, total amount}
//////////////////

Sample User Query: List the names of items ordered by user X.
Can vector database identify the relations between entities automatically?

Thanks.

**Comment by yhmo** [comment]
Hmm..... I think the traditional relation-based database is more suitable for this scenario

**Comment by xiaofan-luan** [comment]
i think the easiest way to do that is use search or SQL2Query

**Comment by codingjaguar** [comment]
Maybe what you want is something that can automatically write SQL for queries expressed by natural language?
See if Text-to-SQL is a promising direction: https://medium.com/@tao.yu/spider-one-more-step-towards-natural-language-interfaces-to-databases-62298dc6df3c
